pilot/cmd/pilot-agent/main.go

```diff
@@ -378,7 +378,7 @@ func getDNSDomain(domain string) string {
 }
 
 func parseApplicationPorts() ([]uint16, error) {
-	parsedPorts := make([]uint16, len(applicationPorts))
+	parsedPorts := []uint16{}
```

Comment:
Actually, I think this should be `make([]uint16, 0, len(applicationPorts))`

Reply:
SGTM